### PR TITLE
feat(useArrayUnique): add a custom function to deduplicate

### PR DESCRIPTION
**Pull request by Alfred-Skyblue**
### Description

thanks @bpolaszek for the https://github.com/vueuse/vueuse/pull/2610 contribution!
I think it seems more appropriate to use a custom function here

---

### What is the purpose of this pull request? 

- [ ] Bug fix
- [x] New Feature
- [x] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [x] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: packages/shared/useArrayUnique/index.md [hunk 1]
```diff
@@ -33,6 +33,24 @@ const list = reactive([1, 2, 2, 3])
 const result = useArrayUnique(list)
 // result.value: [1, 2, 3]
 
-result.value.push(1)
+result.push(1)
```
Comment (antfu) [inline packages/shared/useArrayUnique/index.md:36]: What's the reason of this change?

File: packages/shared/useArrayUnique/index.ts [hunk 2]
```diff
@@ -3,13 +3,32 @@ import { computed } from 'vue-demi'
...
  * reactive unique array
  * @see https://vueuse.org/useArrayUnique
  * @param {Array} list - the array was called upon.
+ * @param fn - the function to compare
  * @returns {Array} A computed ref that returns a unique array of items.
  */
 export function useArrayUnique<T>(
-  list: MaybeComputedRef<MaybeComputedRef<T>[]>): ComputedRef<T[]> {
-  return computed<T[]>(() => [...new Set<T>(resolveUnref(list).map(element => resolveUnref(element)))])
+  list: MaybeComputedRef<MaybeComputedRef<T>[]>, fn?: (a: T, b: T, array: T[]) => boolean): ComputedRef<T[]> {
```
Comment (antfu) [inline packages/shared/useArrayUnique/index.ts:29]:
```suggestion
  list: MaybeComputedRef<MaybeComputedRef<T>[]>, compareFn?: (a: T, b: T, array: T[]) => boolean): ComputedRef<T[]> {
```

File: packages/shared/useArrayUnique/index.md (hunk 1 above)
Comment (Alfred-Skyblue) [reply packages/shared/useArrayUnique/index.md]: Sorry here should be list.push I have modified it in the latest commit

**Review by antfu — APPROVED** [review]
